spec/integration/compose/expected_volumes_spec.rb

```diff
@@ -0,0 +1,31 @@
+require "spec_helper"
+
+RSpec.describe "Expected volumes" do
+  def self.rails_projects
+    rails_service_names = ComposeHelper.rails_app_services.keys
+
+    ProjectsHelper.all_projects.select do |project_name|
+      project_service_names = ComposeHelper.services(project_name).keys
+      rails_service_names.intersection(project_service_names).any?
+    end
+  end
+
+  rails_projects.each do |project_name|
```

Comment:
Minor: you could reduce the nesting around the examples by having the helper return `rails_services`.

Reply:
Do you mean such that the loop is `rails_projects.each do |project_name, service_name, service|` ? 

If so I toyed with it but found it a pain to give the methods a nice name e.g `def services_for_rails_projects` and thought `rails_projects` didn't imply you'd get services. So I'm open to it if you have suggestions of how it can be cleaner.

Reply:
Oh yes, I see what you mean - I didn't clock that `project_name` is needed as well.

I can't think of a clearer way to articulate these helpers as-is. Perhaps something more OO would be the way to go in future, so you could (for example) ask a service object about its name, project and config. Similarly, we could have an object to represent Makefile rules - bet there's a gem for that.

Definitely out-of-scope here!